### PR TITLE
Functionality to delete customer

### LINKED ISSUE #5: backend: ability to delete a customer
DELETE method in controller layer. Use repository.deleteById method with id provided in request as a path variable

### PR DESCRIPTION
**Pull request by ivuorio**
Resolves #5

### CODE REVIEW

File: src/test/kotlin/three/consulting/epoc/repository/CustomerServiceIntegrationTest.kt [hunk 1]
```diff
@@ -68,4 +69,19 @@ class CustomerServiceIntegrationTest : IntegrationTest() {
             .isInstanceOf(UnableToUpdateCustomerException::class.java)
             .hasMessage("Cannot update customer, missing customer id")
     }
+
+    @Test
+    fun `delete customer removes customer from the database`() {
+        assertThat(customerService.findCustomerForId(2L)).isNotNull
+        customerService.deleteCustomer(2L)
+        assertThat(customerService.findCustomerForId(2L)).isNull()
+    }
```
Comment (ivuorio) [inline src/test/kotlin/three/consulting/epoc/repository/CustomerServiceIntegrationTest.kt:78]: Is this overly complicated or would it be enough to just run delete and assert that no errors are thrown?

File: src/main/kotlin/three/consulting/epoc/service/CustomerService.kt [hunk 2]
```diff
@@ -33,6 +34,14 @@ class CustomerService(private val customerRepository: CustomerRepository) {
         else throw UnableToUpdateCustomerException()
     }
 
+    fun deleteCustomer(customerId: Long) {
+        try {
+            customerRepository.deleteById(customerId)
+        }catch (e: EmptyResultDataAccessException) {
```
Comment (LeadingMoominExpert) [inline src/main/kotlin/three/consulting/epoc/service/CustomerService.kt:40]: Missing whitespace before catch, e.g. `} catch (`

File: src/main/kotlin/three/consulting/epoc/service/CustomerService.kt [hunk 3]
```diff
@@ -33,6 +34,14 @@ class CustomerService(private val customerRepository: CustomerRepository) {
...
+            throw UnableToDeleteCustomerException(customerId)
```
Comment (LeadingMoominExpert) [inline src/main/kotlin/three/consulting/epoc/service/CustomerService.kt:41]: When catching an expcetion and throwing a new one, you can include the original in the new one and pass it along to the RunTimeException for improved logging. So:
```kotlin
throw UnableToDeleteCustomerException(customerId, e)
```

File: src/main/kotlin/three/consulting/epoc/service/CustomerService.kt [hunk 4]
```diff
@@ -33,6 +34,14 @@ class CustomerService(private val customerRepository: CustomerRepository) {
...
+        }
+    }
 }
 class UnableToCreateCustomerException : RuntimeException("Cannot create a customer with existing id")
 class UnableToUpdateCustomerException : RuntimeException("Cannot update customer, missing customer id")
+class UnableToDeleteCustomerException(id: Long) : RuntimeException("Cannot delete customer, no customer found for the given id: $id")
```
Comment (LeadingMoominExpert) [inline src/main/kotlin/three/consulting/epoc/service/CustomerService.kt:47]: And here pass the original exception to the RunTimeException
```kotlin
class UnableToDeleteCustomerException(id: Long, exception: EmptyResultDataAccessException) : 
    RuntimeException("Cannot delete customer, no customer found for the given id: $id", exception)
```

File: src/test/kotlin/three/consulting/epoc/controller/CustomerControllerIntegrationTest.kt [hunk 5]
```diff
@@ -54,4 +54,26 @@ class CustomerControllerIntegrationTest : ControllerIntegrationTest() {
...
+        )
+        assertThat(response.statusCode).isEqualTo(HttpStatus.OK)
+    }
+
+    @Test
+    fun `delete without path id return 400`() {
+        val response = restTemplate.exchange(
+            URI("/customer"),
+            HttpMethod.DELETE,
+            jsonPostEntity("src/test/resources/customer/validCreation.json"),
```
Comment (LeadingMoominExpert) [inline src/test/kotlin/three/consulting/epoc/controller/CustomerControllerIntegrationTest.kt:74]: Should be null as the method doesn't accept a requestBody.

File: src/test/kotlin/three/consulting/epoc/repository/CustomerServiceIntegrationTest.kt (hunk 1 above)
Comment (LeadingMoominExpert) [reply src/test/kotlin/three/consulting/epoc/repository/CustomerServiceIntegrationTest.kt:78]: No this is perfectly fine, if you prefer to verify the existence of the customer with said id first.

**Review by LeadingMoominExpert — CHANGES_REQUESTED** [review]

File: src/main/kotlin/three/consulting/epoc/service/CustomerService.kt [hunk 6]
```diff
@@ -33,6 +34,14 @@ class CustomerService(private val customerRepository: CustomerRepository) {
...
+        try {
+            customerRepository.deleteById(customerId)
+        } catch (e: EmptyResultDataAccessException) {
+            throw UnableToDeleteCustomerException(customerId, e)
+        }
+    }
 }
 class UnableToCreateCustomerException : RuntimeException("Cannot create a customer with existing id")
 class UnableToUpdateCustomerException : RuntimeException("Cannot update customer, missing customer id")
+class UnableToDeleteCustomerException(id: Long, exception: EmptyResultDataAccessException) : RuntimeException("Cannot delete customer, no customer found for the given id: $id", exception)
```
Comment (LeadingMoominExpert) [inline src/main/kotlin/three/consulting/epoc/service/CustomerService.kt:47]: Ok, but very long lines like this you should split. Here a newline tight after the `:` would be along ktlint guidelines

**Review by LeadingMoominExpert — APPROVED** [review]

**Pull request merged**